### ISSUE TITLE
Support Asynchronous testing

### ISSUE THREAD

**Issue by aomsweet** [quarkusio/quarkus#24288]
### Description

The current test method is executed synchronously and it cannot test asynchronous methods.  Vertx supports asynchronous testing, but I didn't find a similar way in Quarkus. Am I missing something?
```java
@ExtendWith(VertxExtension.class)
class SomeTest {

  @Test
  @Timeout(value = 10, timeUnit = TimeUnit.SECONDS)
  void some_test(Vertx vertx, VertxTestContext context) {
    // (...)
  }
}
```

**Comment by geoand** [comment]
We already provide https://github.com/quarkusio/quarkus/blob/2.7.4.Final/test-framework/vertx/src/main/java/io/quarkus/test/vertx/RunOnVertxContext.java which is used internally but it's not documented (yet).

Can you have a look at the various tests that use it [here](https://github.com/quarkusio/quarkus/tree/2.7.4.Final/extensions/hibernate-reactive/deployment/src/test/java/io/quarkus/hibernate/reactive) and let us know if it covers what you had in mind?

**Comment by aomsweet** [comment]
Hi, @geoand  Thanks for you reply.  I tried the following way and it works fine.  

```java
    @Test
    @RunOnVertxContext
    void checkPartitions(UniAsserter asserter) {
        Future<Void> future = transferService.checkPartitions();
        Uni<Void> uni = Uni.createFrom().completionStage(future.toCompletionStage());
        asserter.assertNull(() -> uni);
    }
```

Is it possible to provide the `FutureAsserter` interface, because sometimes only Vertx is used, Like the following:

```java
    @Test
    @RunOnVertxContext
    void checkPartitions(FutureAsserter asserter) {
        asserter.assertNull(() -> transferService.checkPartitions());
    }
```

**Comment by geoand** [comment]
Thanks for checking!

What kind of `Future` are you talking about? Vert.x or JDK?

**Comment by aomsweet** [comment]
> 

Future of Vertx. Most of my projects are based on Vertx API, not Mutiny (to be honest, I prefer Vertx's Promise and Future than Mutiny). So I think adding `FutureAsserter` is fine for me. Of course it's up to the Quarkus community.

**Comment by aomsweet** [comment]
> What kind of `Future` are you talking about? Vert.x or JDK?

Future of Vertx. Most of my projects are based on Vertx API, not Mutiny (to be honest, I prefer Vertx's Promise and Future than Mutiny). So I think adding `FutureAsserter` is fine for me. Of course it's up to the Quarkus community.

**Comment by geoand** [comment]
@cescoffier I assume Munity provides some interop with Vert.x's Promise and Future, correct?

**Comment by cescoffier** [comment]
Not really, I would recommend not using them especially if you use Mutiny. Mutiny provides a testing library: https://smallrye.io/smallrye-mutiny/guides/testing, or the UniAsserter. 

It seems that you are still using plain Vert.x API in your application. Mutiny integration goes way beyond API, and many things may not work with the plain Vert.x API.

**Comment by geoand** [comment]
Okay, so I'll close this as adding something like `FutureAsserter` is not something we want to focus on going forward

**Comment by cescoffier** [comment]
BTW: 
```java
        Uni<Void> uni = Uni.createFrom().completionStage(future.toCompletionStage());
```

Is incorrect.

```java
        Uni<Void> uni = Uni.createFrom().completionStage(() -> future.toCompletionStage());
```

is correct.

**Comment by aomsweet** [comment]
ok! I will learn more about Mutiny. Thanks. @cescoffier @geoand

**Comment by geoand** [comment]
YW!
